src/oracles/ChainlinkBoundedCompositeOracle.sol

```diff
@@ -0,0 +1,275 @@
+
+import {AggregatorV3Interface} from "./AggregatorV3Interface.sol";
+
+/// @title ChainlinkBoundedCompositeOracle
+/// @notice Oracle that checks if a primary price feed is within bounds, then either:
+/// - If within bounds: Combines it with a second price feed
+/// - If out of bounds: Falls back to a backup price feed
+/// @dev Includes OEV (Oracle Extractable Value) functionality for early updates
+contract ChainlinkBoundedCompositeOracle is AggregatorV3Interface, Ownable {
+    using Address for address;
+    using SafeCast for *;
+
+    /// @notice Emitted when bounds are updated
+    event BoundsUpdated(
+        int256 oldLower,
+        int256 newLower,
+        int256 oldUpper,
+        int256 newUpper
+    );
+
+    /// @notice Emitted when the fee multiplier is updated
+    event FeeMultiplierUpdated(
+        uint16 oldFeeMultiplier,
+        uint16 newFeeMultiplier
+    );
+
+    /// @notice Emitted when the early update window is updated
+    event EarlyUpdateWindowUpdated(uint256 oldWindow, uint256 newWindow);
+
+    /// @notice Emitted when primary oracle address is updated
+    event PrimaryOracleUpdated(
+        address oldPrimaryOracle,
+        address newPrimaryOracle
+    );
+
+    /// @notice Emitted when BTC oracle address is updated
+    event BTCOracleUpdated(address oldBTCOracle, address newBTCOracle);
+
+    /// @notice Emitted when fallback oracle address is updated
+    event FallbackOracleUpdated(
+        address oldFallbackOracle,
+        address newFallbackOracle
+    );
+
+    /// @notice Primary oracle to check bounds against
+    AggregatorV3Interface public primaryLBTCOracle;
+
+    /// @notice Fallback market rate oracle to use if primary is out of bounds
+    AggregatorV3Interface public fallbackLBTCOracle;
+
+    /// @notice Lower bound for primary oracle price
+    int256 public lowerBound;
+
+    /// @notice Upper bound for primary oracle price
+    int256 public upperBound;
+
+    /// @notice Scaling factor for price calculations
+    uint8 public constant decimals = 18;
+
+    /// @notice constructor to initialize the oracle with price feeds and configuration
+    /// @param _primaryOracle The address of the primary oracle to check bounds against
+    /// @param _fallbackOracle The address of the fallback oracle to use if primary is out of bounds
+    /// @param _lowerBound The lower bound for the primary oracle price
+    /// @param _upperBound The upper bound for the primary oracle price
+    /// @param _governor The address of the governor to own the oracle
+    constructor(
+        address _primaryOracle,
+        address _fallbackOracle,
+        int256 _lowerBound,
+        int256 _upperBound,
+        address _governor
+    ) {
+        primaryLBTCOracle = AggregatorV3Interface(_primaryOracle);
+        fallbackLBTCOracle = AggregatorV3Interface(_fallbackOracle);
+
+        require(
+            _lowerBound < _upperBound,
```

Comment:
it is not an unlikely scenario that one might want to set `_lowerBound = _upperBound = 1`, and currently it is not possible.
Currently it is not our plan, and can also easily workaround it with a 1 wei delta, but consider using `<=` instead of `<`.

Reply:
yes, that the idea